sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -262,6 +244,27 @@ static void initializeIntegrationsAndProcessors(
     if (options.getCompositePerformanceCollector() instanceof NoOpCompositePerformanceCollector) {
       options.setCompositePerformanceCollector(new DefaultCompositePerformanceCollector(options));
     }
+
+    // Check if the profiler was already instantiated in the app start.
+    // We use the Android profiler, that uses a global start/stop api, so we need to preserve the
+    // state of the profiler, and it's only possible retaining the instance.
+    final @NotNull AppStartMetrics appStartMetrics = AppStartMetrics.getInstance();
```

Comment:
I'm a little bit worried about moving this down here - there's some stuff happening in the `AppStartMetrics` constructor which may affect some other parts. Can we keep this line as it was before and only move the profiler instantiating down?